include/openssl/rsa.h

```diff
@@ -192,15 +202,22 @@ int RSA_security_bits(const RSA *rsa);
 int RSA_set0_key(RSA *r, BIGNUM *n, BIGNUM *e, BIGNUM *d);
 int RSA_set0_factors(RSA *r, BIGNUM *p, BIGNUM *q);
 int RSA_set0_crt_params(RSA *r,BIGNUM *dmp1, BIGNUM *dmq1, BIGNUM *iqmp);
+int RSA_set0_multi_prime_params(RSA *r, BIGNUM **primes, BIGNUM **exps,
+                                BIGNUM **coeffs, size_t pnum);
 void RSA_get0_key(const RSA *r,
                   const BIGNUM **n, const BIGNUM **e, const BIGNUM **d);
 void RSA_get0_factors(const RSA *r, const BIGNUM **p, const BIGNUM **q);
+size_t RSA_get_multi_prime_extra_count(const RSA *r);
+int RSA_get0_multi_prime_factors(const RSA *r, const BIGNUM **oth_primes);
 void RSA_get0_crt_params(const RSA *r,
                          const BIGNUM **dmp1, const BIGNUM **dmq1,
                          const BIGNUM **iqmp);
+int RSA_get0_multi_prime_crt_params(const RSA *r, const BIGNUM **oth_exps,
+                                     const BIGNUM **oth_coeffs);
 void RSA_clear_flags(RSA *r, int flags);
 int RSA_test_flags(const RSA *r, int flags);
 void RSA_set_flags(RSA *r, int flags);
+int32_t RSA_get_version(RSA *r);
```

Comment:
The earler suggestion to adhere to base types in order to minimize dependencies on other headers was rather call for broader discussion. I mean I'd like to hear what would more people think...

Reply:
I would stick to the base types unless there is a strong reason to not do so. 
And this does not look like a reason.
So I would use 'int' here both as return and as parameter type.

Reply:
So this is not a blocker for this feature, right?

Reply:
> So I would use 'int' here both as return and as parameter type.

There is no parameter anymore, return type is the only one left. And it's sufficient to just change prototype and rely on [no-op] implicit cast.

Reply:
changed to int